lib/layer.js

```diff
@@ -28,23 +28,32 @@ var hasOwnProperty = Object.prototype.hasOwnProperty
 
 module.exports = Layer
 
-function Layer(path, options, fn) {
+function Layer(paths, options, fn) {
   if (!(this instanceof Layer)) {
-    return new Layer(path, options, fn)
+    return new Layer(paths, options, fn)
   }
 
-  debug('new %s', path)
+  debug('new %s', paths)
   var opts = options || {}
 
   this.handle = fn
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
-  this.regexp = pathRegexp(path, this.keys = [], opts)
 
-  if (path === '/' && opts.end === false) {
-    this.regexp.fast_slash = true
+  if (paths === '/' && opts.end === false) {
+    this.fastSlash = true
```

Comment:
V8 can do optimizations if the shape of an object remains the same.  This addition will have only some `Layer`'s with a `fastSlash` property.  To optimize this, you should define it as `undefined` above, or `false` here.

Reply:
Fixed in https://github.com/pillarjs/router/pull/86/files#diff-be883fbed8968009576c828e0ff70e36R47